src/app/shared/recognize-video-type-id.ts

```diff
@@ -0,0 +1,199 @@
+
+const getSrc = require('get-src');
+
+export function getVideoId(str: any): any {
+  if (typeof str !== 'string') {
+    throw new TypeError('get-video-id expects a string');
```

Comment:
to właściwie będzie zbędne w ts, bo jak opiszesz parametr że musi być stringiem to nikt tu nie poda niestringa (to właśnie przewaga ts nad jsem, wiadomo jakie są typy argumentów i zwrotek)